src/app/slash_plugins/dog.plugin.ts

```diff
@@ -0,0 +1,151 @@
+import { MessageEmbed } from 'discord.js';
+import { ISlashCommand } from '../../common/slash';
+import { IContainer, IHttpResponse } from '../../common/types';
+
+const API_URL: string = 'https://dog.ceo/api/';
+
+let allBreeds: Set<string> = new Set([]);
+let breeds: string[] = [];
+let subBreeds: IDogSubBreed[] = [];
+
+const plugin: ISlashCommand = {
+  commandName: 'dog',
+  name: 'Dog Plugin',
+  description: 'Generates pictures of doggos.',
+  options: [
+    {
+      name: 'breed',
+      description: 'The breed of dog to generate. Or "list" to list all breeds.',
+      type: 'STRING',
+      required: false,
+    },
+  ],
+  async execute({ interaction, container }) {
+    if (allBreeds.size === 0) {
```

Comment:
Discord commands by default only have a window of 3 seconds to respond or else the response becomes invalid. In the case of this plugin, due to unexpected network latency, we may not be able to guarantee this. Instead we can defer `await interaction.defer()` to tell discord to extend the response window to 15 minutes.

You'll also need to update your `interaction.reply`s to `interaction.followUp` since we're telling discord we're updating a deferred interaction rather than a normal one.

```suggestion
  async execute({ interaction, container }) {
  	await interaction.defer();
    if (allBreeds.size === 0) {
```

Reply:
Ok awesome good to know